presto-main-base/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2668,6 +2727,7 @@ private Scope computeAndAssignOutputScope(QuerySpecification node, Optional<Scop
                     Optional<QualifiedName> starPrefix = ((AllColumns) item).getPrefix();
 
                     for (Field field : sourceScope.getRelationType().resolveFieldsWithPrefix(starPrefix)) {
+                        analysis.addSourceColumns(field, analysis.getSourceColumns(field));
```

Comment:
This line appears to be adding source columns from a field back to itself, which seems redundant. Consider removing this line or clarifying its purpose.